src/python/WMCore/MicroService/Unified/MSTransferor.py

```diff
@@ -326,3 +360,105 @@ def createTransferDoc(self, reqName, transferRecords):
         msg += "Error: %s" % resp
         self.logger.error(msg)
         return False
+
+
+    def _getStorageQuota(self):
+        """
+        Fetch the DataOps quota from Detox. At this stage, we do not do
+        any manipulation with the quota value (Unified uses 80% of the quota),
+        use it as is!
+
+        :return: create an instance cache structure to keep track of quota
+          and available storage. The structure is as follows:
+          {"pnn_name": {"quota": quota in bytes for the rucio account or phedex group,
+                        "bytes_limit": total space for the account/group,
+                        "bytes": amount of bytes currently used/archived,
+                        "bytes_remaining": space remaining for the acct/group,
+                        "quota_avail": a fraction of the quota that we will use}
+
+        NOTE: code extracted/modified from Unified, see `fetch_detox_info` in
+          https://github.com/CMSCompOps/WmAgentScripts/blob/master/utils.py#L2514
+        """
+        # FIXME: extremely fragile code that has to be replaced by a proper
+        # CRIC/Rucio API in the very near future
+        info = getDetoxQuota(self.msConfig['detoxUrl'])
+
+        doRead = False
+        for line in info:
+            if 'DDM Partition:' in line and self.msConfig['quotaAccount'] in line:
+                doRead = True
+                continue
+            elif 'DDM Partition:' in line:
+                doRead = False
+                continue
+            elif line.startswith('#'):
+                continue
+
+            if not doRead:
+                continue
+
+            _, quota, _, _, pnn = line.split()
+
+            if pnn.endswith("_MSS") or pnn.endswith("_Export"):
+                continue
+            # convert from TB to bytes
+            self.nodesUsage.setdefault(pnn, {})
+            self.nodesUsage[pnn] = dict(quota=int(quota) * (1024 ** 4),
+                                        bytes_limit=0,
+                                        bytes=0,
+                                        bytes_remaining=0,
+                                        quota_avail=0)
+
+
+    def _getStorageUsage(self):
```

Comment:
This method is not thread safe either. Please convert it to accept nodeUsage dict and return an updated version.